productivity-suite/app/legacy-app/src/auth.tsx

```diff
@@ -3,44 +3,52 @@ import { createContext, useContext, useEffect, useState } from "react";
 import { Navigate, useLocation, useNavigate } from "react-router-dom";
 import {
   deleteCurrentUser,
+  getCurrentUser,
   getUserData,
   saveCurrentUser,
   setUserData,
 } from "shared";
 import { initialPlaceholderTodoList } from "./initialPlaceholderTodoList";
 
-interface AuthContextType {
-  user: string | null;
-  login: (userName: string) => void;
-  logout: () => void;
-}
+export const AUTH_LOADING = Symbol("Auth Loading");
+export type AuthContextType = { user: string | null } | undefined;
 
 let AuthContext = createContext<AuthContextType>(null!);
 
 export function AuthProvider({ children }: { children: React.ReactNode }) {
   const navigate = useNavigate();
+  const location = useLocation();
+
+  const [user, setUser] = useState<string | null | typeof AUTH_LOADING>(
+    AUTH_LOADING
+  );
 
-  const [user, setUser] = useState<string | null>(null);
   useEffect(() => {
-    return getBus().listen<{ username: string }>("authentication", (auth) => {
-      setUser(auth?.username);
-    });
-  });
+    getCurrentUser().then((username) => setUser(username));
```

Comment:
would this look nicer with async/await?

```ts
useEffect(async () => setUser(await getCurrentUser()));
```

Reply:
this is getting the user from the cookie and ignoring the one present in the message bus

I think we should get the data from the message bus, shouldn't we?

(
and if we don't I think we should clean this up?
https://github.com/cloudflare/workers-web-experiments/blob/56a556a31d03428ea503721d3636b8dc0aa38487/productivity-suite/app/legacy-app/src/worker/script.ts#L25-L29
)

Reply:
Can't pass async functions to `useEffect`

Reply:
You can't pass `async` functions to `useEffect`

Reply:
ahhh, ok sorry my bad 😅

Reply:
by the way, I still stand with my comment that I think the data should come from the bus 😜

Reply:
You are right, we can indeed just use the message bus to get the initial value.
This messed with my brain when we were working on this because I couldn't work out where this would be initialized.
But I think it is always done in the Worker, right?

Anyway, I changed it and all seems well 🙏

Reply:
Thanks 😄 

Yeah it should be all done by the piercing gateway, at the beginning of the request handling we set the value using the user provided `generateMessageBusState`:

https://github.com/cloudflare/workers-web-experiments/blob/0da0c31dfce27e8ae5c992cb57565efddd106cd6/productivity-suite/app/legacy-app/src/worker/script.ts#L21-L29

(and the message bus is applied by the piercing gateway as it fetches the index.html:
https://github.com/cloudflare/workers-web-experiments/blob/0da0c31dfce27e8ae5c992cb57565efddd106cd6/productivity-suite/piercing-library/src/piercing-gateway.ts#L310)